### PR TITLE
Update to react-native@0.72.5

### PR DESCRIPTION
**Pull request by pfrazee**
Closes https://github.com/bluesky-social/social-app/issues/1594 by pulling in the bugfix highlighted below:

CleanShot 2023-10-03 at 16 43 26@2x

### CODE REVIEW

File: patches/react-native+0.72.5.patch [hunk 1]
```diff
@@ -0,0 +1,1787 @@
...
+   }
+-  _lastStringStateWasUpdatedWith = _backedTextInputView.attributedText;
+   [self textViewProbablyDidChangeSelection];
+ }
+ 
+diff --git a/node_modules/react-native/ReactCommon/react/renderer/components/rncore/ComponentDescriptors.h b/node_modules/react-native/ReactCommon/react/renderer/components/rncore/ComponentDescriptors.h
+new file mode 100644
+index 0000000..37b7d8d
+--- /dev/null
++++ b/node_modules/react-native/ReactCommon/react/renderer/components/rncore/ComponentDescriptors.h
```
Comment (gaearon) [inline patches/react-native+0.72.5.patch:22]: What's up with this patch? Is this intentional? The old patch was way smaller

Comment (pfrazee) [reply patches/react-native+0.72.5.patch:22]: Oh weird, no. `patch-package` must have bugged out with its process for updating the patch. Glad you caught this, I'll dig into it.

**Review by gaearon — APPROVED** [review]
assuming you'll fix the patch

File: patches/react-native+0.72.5.patch (hunk 1 above)
Comment (pfrazee) [reply patches/react-native+0.72.5.patch]: Fixed

**Pull request merged**